pict-lib/pict/private/pict.rkt

```diff
@@ -792,14 +927,23 @@
                 cc-superimpose
                 ctl-superimpose
                 cbl-superimpose)
-  (let ([make-superimpose 
+  (let ([make-superimpose
          (lambda (get-h get-v get-th name)
-           (lambda boxes
-             (when (null? boxes)
+           (lambda boxes*
+             (when (null? boxes*)
                (error name "expected at least one argument, got none"))
-             (unless (andmap pict? boxes)
-               (error name "expected all picts as arguments, got ~a"
-                      (apply string-append (add-between (map (λ (x) (format "~e" x)) boxes) " "))))
+             (define boxes
+               (map
+                (lambda (p)
+                  (cond
+                    [(pict? p) p]
+                    [(pict-convertible? p)
+                     (pict-convert p)]
+                    [else
+                     (raise-argument-error
+                      name "all picts as arguments"
+                      (apply string-append (add-between (map (λ (x) (format "~e" x)) boxes*) " ")))]))
```

Comment:
I think you want (apply raise-argument-error name "pict?" i ...) where the ellipses are filled in with a list of all of the arguments to the original function and "i" is the number of the argument you're looking at.

Or put a contract? (Or maybe there is one already?)